### PR TITLE
Pagination sorting dev

### PR DESCRIPTION
**Pull request by don-bigdad**

### CODE REVIEW

File: src/main/java/com/example/springbootbookshop/controller/BookController.java [hunk 1]
```diff
@@ -4,33 +4,65 @@
...
 @RestController
+@Validated
 @RequestMapping(value = "/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books from DB")
```
Comment (dmytrocherepov) [inline src/main/java/com/example/springbootbookshop/controller/BookController.java:36]: Maybe you should add description to your Operations , but i suppose that it is up you

**Review by dmytrocherepov — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -20,6 +20,16 @@
         <mapstruct.version>1.5.5.Final</mapstruct.version>
     </properties>
     <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-validation</artifactId>
+            <version>3.1.4</version>
```
Comment (NazarSmith228) [inline pom.xml]: you may skip version here

File: src/main/java/com/example/springbootbookshop/controller/BookController.java [hunk 3]
```diff
@@ -4,33 +4,65 @@
...
+    @Operation(summary = "Get book by Id from DB")
+    @ResponseStatus(HttpStatus.OK)
+    public BookDto getBookById(@PathVariable @Positive Long id) {
         return bookService.getBookById(id);
     }
 
+    @PutMapping(value = "/{id}")
+    @Operation(summary = "Update book by Id in DB")
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public BookDto getBookById(@RequestBody @Valid CreateBookRequestDto bookDto,
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/controller/BookController.java]: is it really `get`?)

File: src/main/java/com/example/springbootbookshop/controller/BookController.java [hunk 4]
```diff
@@ -4,33 +4,65 @@
...
 @RequestMapping(value = "/books")
 public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books from DB")
+    @ResponseStatus(HttpStatus.OK)
+    public List<BookDto> getAll(Pageable pageable) {
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/controller/BookController.java]: makes sense to add `@PageableDefault`

File: src/main/java/com/example/springbootbookshop/controller/BookController.java [hunk 5]
```diff
@@ -4,33 +4,65 @@
...
+        return bookService.findAll(pageable);
+    }
+
+    @DeleteMapping(value = "/{id}")
+    @Operation(summary = "Delete book by id")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteBookById(@PathVariable Long id) {
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/controller/BookController.java]: you've added `@Validated` but you don't validate `@PathVariable` :)

```suggestion
    public void deleteBookById(@PathVariable @Positive Long id) {
```

File: src/main/java/com/example/springbootbookshop/exception/CustomGlobalExceptionHandler.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body,headers,status);
+    }
+
+    @ExceptionHandler(EntityNotFoundException.class)
+    public ResponseEntity<Object> handleEntityNotFoundException(
+            EntityNotFoundException ex) {
+        System.out.println(ex.getMessage());
+        return new ResponseEntity<>(ex.getLocalizedMessage(), HttpStatus.BAD_REQUEST);
```
Comment (NazarSmith228) [inline src/main/java/com/example/springbootbookshop/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(ex.getLocalizedMessage(), HttpStatus.NOT_FOUND);
```

**Review by NazarSmith228 — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution

**Review by NazarSmith228 — APPROVED** [review]
LGTM

**Pull request merged**